src/adv.py

```diff
@@ -49,3 +56,28 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+while True:
```

Comment:
Nice job here abstracting nested if/elif/else blocks into functions and keeping this while loop clean.